sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -161,6 +161,9 @@ statement
     | ALTER TABLE table=multipartIdentifier
         (ALTER | CHANGE) COLUMN? column=multipartIdentifier
         (TYPE dataType)? (COMMENT comment=STRING)? colPosition?        #alterTableColumn
+    | ALTER TABLE table=multipartIdentifier partitionSpec?
+        CHANGE COLUMN?
+        colName=multipartIdentifier colType colPosition?               #hiveChangeColumn
```

Comment:
`colName` was `errorCapturingIdentifier` before. We want it to be `multipartIdentifier`?

Reply:
Though it is fine because `ResolveSessionCatalog` will detect and throw `AnalysisException("ALTER COLUMN with qualified column is only supported with v2 tables.")`.

Reply:
This is to be consistent with the other ALTER COLUMN syntax.